ament_cmake_xmllint/cmake/ament_cmake_xmllint_lint_hook.cmake

```diff
@@ -12,8 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-file(GLOB_RECURSE _source_files FOLLOW_SYMLINKS "*.xml")
+# Forces ament_xmllint to consider ament_cmake_xmllint_EXTENSIONS as the given extensions if defined
+set(_extensions "xml")
```

Comment:
This feels a little weird.  If the user doesn't specify anything for EXTENSIONS, then we won't pass `--extensions` to `ament_xmllint` at all, and then that script will choose the `xml` extension itself.  If the user specifies some EXTENSIONS, then we will always pass `--extensions xml`, plus whatever they choose.  There is no way for the user to ask for extensions that *doesn't* include `xml`.

I think we should do one of two things here:
1.  If the user specifies EXTENSIONS, then make that the complete set.  Don't silently add `.xml`.
2.  Rename the variable from EXTENSIONS to EXTENSIONS_APPEND or something like that, to make it clear that we will always do at least xml.

Either way, we should document what we are doing in the documentation to `ament_xmllint`.

Reply:
No, I checked the behaviour. When setting `ament_cmake_xmllint_EXTENSIONS`, it will completely overwrite the `_extensions` variable. As I use `REGEX REPLACE`.